Bottom: 0da612e7db00d5a0d5de8140d348a69e115ff92a
Top:    63d3bf9e2dfd8125193799f1932cfb1a832edae3
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-02-20 19:43:38 +0800

Refresh of virtio-net-optimize-small

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 1c97877..b07c4b0 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -734,7 +734,7 @@ static void receive_buf(struct virtnet_info *vi, struct receive_queue *rq,
 		} else if (vi->big_packets) {
 			give_pages(rq, buf);
 		} else {
-			dev_kfree_skb(buf);
+			put_page(virt_to_head_page(buf));
 		}
 		return;
 	}
